code.gs

```diff
@@ -2333,3 +2355,30 @@ function notifStatus () { // eslint-disable-line no-unused-vars
   log.add(toLog);
   log.sendEmail(settings.user.notificationEmail, settings.user.emailSenderName);
 }
+
+/**
+ * Generate a date with a given timezone id.
+ *
+ * @param {!Number} year - Full year of the date.
+ * @param {!Number} month - Month of the date, starting from 0.
+ * @param {!Number} day - Day of the date, starting from 1.
+ * @param {!Number} hour - Hour of the date.
+ * @param {!Number} minute - Minute of the date.
+ * @param {!Number} second - Second of the date,
+ * @param {String} timezoneId - A valid IANA timezone identifier.
+ *
+ * @returns {Date} - The date corresponding to the input.
+ */
+function dateWithTimezone (year, month, day, hour, minute, second, timezoneId) {
+  var date, offset;
+
+  // Generate the date as in the UTC0 timezone.
+  date = new Date(Date.UTC(year, month, day, hour, minute, second));
+  // Calculate the offset for the given timezone.
+  offset = Utilities.formatDate(date, timezoneId, 'Z');
+  // Evaluate the offset (in minutes).
+  offset = (offset[0] === '-' ? -1 : +1) * (parseInt(offset[1] + offset[2], 10) * 60 + parseInt(offset[3] + offset[4], 10));
+  // Apply the offse to the UTC date to get the correct date.
+  date = new Date(date.getTime() - offset * 60 * 1000);
```

Comment:
nitpick: This is the same nitpick as I commented on above: `... * 60 * 1000` vs. `... * 60000`. Ignore it if you prefer.